doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,255 +870,297 @@ Module system
 
-   Defines a module type :token:`ident` equal to :token:`module_type`.
+   Similar to :cmd:`Import`, except that when the module containing this command
+   is imported, the :n:`{+ @qualid }` are imported as well.
 
-    .. cmdv:: Module Type @ident {* @module_binding} := @module_type
+   .. exn:: @qualid is not a module.
+      :undocumented:
 
-       Defines a functor type :token:`ident` specifying functors taking arguments :token:`module_bindings` and
-       returning :token:`module_type`.
+   .. warn:: Trying to mask the absolute name @qualid!
+      :undocumented:
 
-    .. cmdv:: Module Type @ident {* @module_binding} := {+<+ @module_type }
+.. cmd:: Print Module @qualid
 
-       is equivalent to an interactive module type were each :token:`module_type` is included.
+   Prints the module type and (optionally) the body of the module :n:`@qualid`.
 
-.. cmd:: Declare Module @ident : @module_type
+.. cmd:: Print Module Type @qualid
 
-   Declares a module :token:`ident` of type :token:`module_type`.
+   Prints the module type corresponding to :n:`@qualid`.
 
-    .. cmdv:: Declare Module @ident {* @module_binding} : @module_type
+.. flag:: Short Module Printing
 
-       Declares a functor with parameters given by the list of :token:`module_binding` and output module type
-       :token:`module_type`.
+   This flag (off by default) disables the printing of the types of fields,
+   leaving only their names, for the commands :cmd:`Print Module` and
+   :cmd:`Print Module Type`.
 
-.. example::
+.. _module_examples:
 
-    Let us define a simple module.
+Examples
+~~~~~~~~
 
-    .. coqtop:: all
+.. example:: Defining a simple module interactively
 
-       Module M.
+    .. coqtop:: in
 
+       Module M.
        Definition T := nat.
-
        Definition x := 0.
 
-       Definition y : bool.
+    .. coqtop:: all
 
+       Definition y : bool.
        exact true.
 
-       Defined.
+    .. coqtop:: in
 
+       Defined.
        End M.
 
-Inside a module one can define constants, prove theorems and do any
-other things that can be done in the toplevel. Components of a closed
+Inside a module one can define constants, prove theorems and do anything
+else that can be done in the toplevel. Components of a closed
 module can be accessed using the dot notation:
 
 .. coqtop:: all
 
    Print M.x.
 
-A simple module type:
-
-.. coqtop:: all
-
-   Module Type SIG.
-
-   Parameter T : Set.
+.. _example_def_simple_module_type:
 
-   Parameter x : T.
+.. example:: Defining a simple module type interactively
 
-   End SIG.
+   .. coqtop:: in
 
-Now we can create a new module from M, giving it a less precise
-specification: the y component is dropped as well as the body of x.
+      Module Type SIG.
+      Parameter T : Set.
+      Parameter x : T.
+      End SIG.
 
-.. coqtop:: all
+.. _example_filter_module:
 
-   Module N : SIG with Definition T := nat := M.
+.. example Creating a new module that omits some items from an existing module
```

Comment:
```suggestion
.. example:: Creating a new module that omits some items from an existing module
```

Reply:
Yup.  Maybe worth creating a script that would grep for problems like this, for example reporting instances of `\.\. example[^:]`.